packages/react-dom/src/__tests__/ReactEmptyComponent-test.js

```diff
@@ -234,44 +245,51 @@ describe('ReactEmptyComponent', () => {
           // Make sure the DOM node resolves properly even if we're replacing a
           // `null` component
           expect(ReactDOM.findDOMNode(this)).not.toBe(null);
-          assertions++;
         }
 
         componentWillUnmount() {
           // Even though we're getting replaced by `null`, we haven't been
           // replaced yet!
           expect(ReactDOM.findDOMNode(this)).not.toBe(null);
-          assertions++;
         }
       }
 
-      class Wrapper extends React.Component {
-        render() {
-          return this.props.showInner ? <Inner /> : nullORUndefined;
-        }
+      function Wrapper({showInner}) {
+        innerRef = React.createRef(null);
+        return showInner ? <Inner ref={innerRef} /> : nullORUndefined;
       }
 
       const el = document.createElement('div');
-      let component;
 
       // Render the <Inner /> component...
-      component = ReactDOM.render(<Wrapper showInner={true} />, el);
-      expect(ReactDOM.findDOMNode(component)).not.toBe(null);
+      const root = ReactDOMClient.createRoot(el);
+      await act(() => {
+        root.render(<Wrapper showInner={true} />);
+      });
+      expect(innerRef.current).not.toBe(null);
 
       // Switch to null...
-      component = ReactDOM.render(<Wrapper showInner={false} />, el);
-      expect(ReactDOM.findDOMNode(component)).toBe(null);
+      await act(() => {
+        root.render(<Wrapper showInner={false} />);
+      });
+      expect(innerRef.current).toBe(null);
 
       // ...then switch back.
-      component = ReactDOM.render(<Wrapper showInner={true} />, el);
-      expect(ReactDOM.findDOMNode(component)).not.toBe(null);
+      await act(() => {
+        root.render(<Wrapper showInner={true} />);
+      });
+      expect(innerRef.current).not.toBe(null);
 
-      expect(assertions).toBe(3);
+      expect.assertions(6);
     });
```

Comment:
There are 6?

Reply:
Yes. One for each of 3 renders + one null check after each render. Before assertions was a counter that was only incremented within render()